[Docs] Reset Docs Colors results in Unexpected Error splash page

<!--- Provide a general summary of the issue in the Title above -->
At https://material-ui.com/customization/color/#color-tool, clicking on the button "Reset Docs Colors" results in a white splash page with the text `An unexpected error has occurred.` I have tested in Chrome and Firefox.

The error in the console:
```
commons.e98f46a2b39a45ee03e2.js:1 Error: Unrecognized type RESET_COLORS
    at commons.e98f46a2b39a45ee03e2.js:1
    at Object.ra [as useReducer] (commons.e98f46a2b39a45ee03e2.js:1)
    at Object.useReducer (commons.e98f46a2b39a45ee03e2.js:1)
    at k (commons.e98f46a2b39a45ee03e2.js:1)
    at Jo (commons.e98f46a2b39a45ee03e2.js:1)
    at Fa (commons.e98f46a2b39a45ee03e2.js:1)
    at za (commons.e98f46a2b39a45ee03e2.js:1)
    at Ji (commons.e98f46a2b39a45ee03e2.js:1)
    at Qi (commons.e98f46a2b39a45ee03e2.js:1)
    at Iu (commons.e98f46a2b39a45ee03e2.js:1)
```

<!--
    Thank you very much for contributing to Material-UI by creating an issue! ❤️
    To avoid duplicate issues we ask you to check off the following list.
-->

<!-- Checked checkbox should look like this: [x] -->
- [x] This is not a v0.x issue. <!-- (v0.x is no longer maintained) -->
- [x] I have searched the [issues](https://github.com/mui-org/material-ui/issues) of this repository and believe that this is not a duplicate.

## Expected Behavior 🤔
<!---
    Describe what should happen.
-->
Clicking on "Reset Docs Colors" results in setting the website to the default theme.

## Your Environment 🌎
<!---
    Include as many relevant details about the environment with which you experienced the bug.
    If you encounter issues with typescript please include version and tsconfig.
-->

| Tech         | Version |
|--------------|---------|
| Chrome      | 75.0.3770.142 (Official Build) (64-bit) |
| Firefox Quantum | 67.0.4 (64-bit) |

--- a/docs/src/modules/components/ThemeContext.js
+++ b/docs/src/modules/components/ThemeContext.js
@@ -123,6 +123,11 @@ export function ThemeProvider(props) {
           dense: themeInitialOptions.dense,
           spacing: themeInitialOptions.spacing,
         };
+      case 'RESET_COLORS':
+        return {
+          ...state,
+          paletteColors: themeInitialOptions.paletteColors,
+        };
       case 'CHANGE':
         return {
           ...state,

